src/inference/src/dev/iplugin.cpp

```diff
@@ -145,6 +205,18 @@ std::unordered_set<std::string> ov::get_supported_nodes(
         });
     };
 
+    auto filter_shapeOf = [&]() {
+        for (auto& op : model->get_ordered_ops()) {
+            const auto& name = op->get_friendly_name();
+            if (ov::is_type<ov::op::util::ShapeOfBase>(op) && (supported.count(name) || removed_nodes.count(name))) {
+                if (has_all_consumers_unsupported(supported, op) && has_all_consumers_unsupported(removed_nodes, op)) {
+                    remove_op_from_supported(op);
+                    removed_nodes.erase(name);
+                }
+            }
+        }
+    };
```

Comment:
can we enable test for the shape of logic? has_all_consumers_unsupported(removed_nodes, op), when do we need to enable check of shape of with removed_nodes?

Reply:
Yes, will add test case for this.